app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,69 @@
+class TripsController < ApplicationController
+  def show
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil? || @trip.driver.nil? || @trip.passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new 
+    if params[:passenger_id]
+      passenger = Passenger.find_by(id: params[:passenger_id])
+      @trip = passenger.trips.new
```

Comment:
Consider passing these parameters to @trip.new in a hash to increase readability.